server/request.go

```diff
@@ -0,0 +1,206 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/asdine/storm"
+	"github.com/go-chi/chi"
+	"github.com/kochman/hotshots/log"
+)
+
+/*
+ * Response Structs
+ */
+
+type GetPhotoIDsResponse struct {
+	Success bool
+	IDs     []string
+}
+
+type GetPhotosResponse struct {
+	Success bool
+	Photos  []Photo
+}
+
+type PostPhotoResponse struct {
+	Success bool
+	NewID   string `json:",omitempty"`
+}
+
+type GetPhotoMetadataResponse struct {
+	Success bool
+	Photo   Photo
+}
+
+type ErrorResponse struct {
+	Success bool
+	Error   string
+}
+
+type NotFoundResponse struct {
+	Success bool
+	Error   string
+	URI     string
+}
```

Comment:
We likely want to `snake_case` all returned JSON fields with some struct tags, but it doesn't matter all that much.